frontend/src/components/pages/CourseDashboard.js

```diff
@@ -24,21 +69,95 @@ export function CourseDashboard() {
             .catch((error) => setError(error.message));
     }, []);
 
+    useEffect(() => {
+        const userId = Cookies.get('userId');
+
+        if (!userId) {
+            setError('User ID not found');
+            return;
+        }
+
+        fetch(`http://localhost:4000/user/${userId}`)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error('Network response was not ok');
+                }
+                return response.json();
+            })
+            .then((data) => setUserInfo(data.user))
+            .catch((error) => setError(error.message));
+    }, []);
+
     if (error) return <p>Error: {error}</p>;
     if (!courseInfo) return <p>Loading...</p>;
+    if (!userInfo) return <p>Loading...</p>;
 
     var courseList = [];
     //Push all courses into courseList
-    courseInfo.forEach(course => {
+    if (userInfo.role === "educator"){
+        courseInfo.forEach(course => {
+            courseList.push(
+                <a href={`/courses/${course.ID}`}>
+                    <div className="bg-gray-100 p-4 rounded shadow hover:bg-gray-300" >
+                        <h3 className="text-xl font-semibold" >{course.title}</h3>
+                        <p className="mt-2">{course.description}</p>
+                    </div>
+                </a>
+            )
+        });
+        //Push the create course button at the end of courses
         courseList.push(
-            <a href={`/courses/${course.ID}`}>
-                <div className="bg-gray-100 p-4 rounded shadow hover:bg-gray-300" >
-                    <h3 className="text-xl font-semibold" >{course.title}</h3>
-                    <p className="mt-2">{course.description}</p>
-                </div>
-            </a>
+            <Popup trigger={<div className="bg-blue-500 p-4 rounded shadow hover:bg-blue-700 m-auto text-center text-xl text-white font-semibold hover:cursor-pointer" > 
```

Comment:
I like this popup window. I would have the corner of the modal a little more rounded rather than squared, and have a little more padding for the inner content of the modal. We could make it a more generic modal as well (AKA a component) that we could reuse for maybe updating user information, etc.

Reply:
I removed the library that I added to make the modal and made it from scratch so it would be more customizable. I also added the style changes you suggested and made it a reusable component in components/Modal.js.